website/src/docs/hotchocolate/fetching-data/resolvers.md

```diff
@@ -0,0 +1,524 @@
+          foo: String!
+        }
+    ")
+    .AddResolver("Query", "foo", (context) => "Bar");
+```
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+## Async Resolver
+
+Most data fetching operations, like calling a service or communicating with a database, will be asynchronous.
+
+In Hot Chocolate we can simply mark our resolver methods and delegates as `async` or return a `Task<T>` and everything works out of the box.
+
+We can also add a `CancellationToken` as argument to our resolver. Hot Chocolate will automatically cancel this token, if the request has been aborted.
+
+```csharp
+public class Query
+{
+    public async Task<string> Foo(CancellationToken ct)
+    {
+        // Omitted code for brevity
+    }
+}
+```
+
+When using a delegate resolver, the `CancellationToken` is passed as second argument to the delegate.
+
+```csharp
+descriptor
+    .Field("foo")
+    .Resolve((context, ct) =>
+    {
+        // Omitted code for brevity
+    });
+```
+
+The `CancellationToken` can also be accessed through the `IResolverContext`.
+
+```csharp
+descriptor
+    .Field("foo")
+    .Resolve(context =>
+    {
+        CancellationToken ct = context.RequestAborted;
+
+        // Omitted code for brevity
+    });
+```
+
+## ResolveWith
+
+Thus far we have looked at two ways to specify resolvers in Code-first:
+
+- Add new methods to the CLR type, e.g. the `T` type of `ObjectType<T>`
+- Add new fields to the schema type in the form of delegates
+  ```csharp
+  descriptor.Field("foo").Resolve(context => )
+  ```
+
+But there's a third way. We can describe our field using the `descriptor`, but instead of a resolver delegate, we can point to a method on another class, responsible for resolving this field.
+
+```csharp
+public class FooResolvers
+{
+    public string GetFoo(string arg, [Service] FooService service)
+    {
+        // Omitted code for brevity
+    }
+}
+
+public class QueryType : ObjectType
+{
+    protected override void Configure(IObjectTypeDescriptor descriptor)
+    {
+        descriptor
+            .Field("foo")
+            .Argument("arg", a => a.Type<NonNullType<StringType>>())
+            .ResolveWith<FooResolvers>(r => r.GetFoo(default, default));
+    }
+}
+```
+
+# Arguments
+
+We can access the arguments we defined in the schema in our resolver, like regular arguments of a function.
+
+There are also specific arguments that will be automatically populated by Hot Chocolate, when the resolver is executed. These include [Dependency injection services](#injecting-services), [DataLoaders](/docs/hotchocolate/fetching-data/dataloader), state, or even context like a [_parent_](#accessing-parent-values) value.
+
+[Learn more about arguments](/docs/hotchocolate/defining-a-schema/arguments)
+
+# Injecting Services
+
+Resolvers integrate nicely with `Microsoft.Extensions.DependecyInjection`.
+We can access all registered services in our resolvers.
+
+Let's assume we have created a `UserService` and registered it as a service.
```

Comment:
I think it's worth putting some words about class-level/ctor-injection vs method-level injection. About how resolver classes / extensions are, by default, registered as singleton as the schema is built up as singleton for perf. 

The impact being that you can't use a ctor to get a scoped/transient service and must use method-level injection.

If you're confident a service will always be singleton, you can use ctor injection.